File: backend/src/timeflow/infrastructure/websocket/messages/session.py
"""连接建立握手消息。"""

from typing import Literal

from pydantic import BaseModel

from timeflow.infrastructure.websocket.messages.envelope import ErrorDetail


class SessionHello(BaseModel):
    """客户端连接后发送的握手消息。"""

    type: Literal["session.hello"] = "session.hello"
    device_id: str
    app_version: str


class SessionReady(BaseModel):
    """服务端确认会话建立成功。"""

    type: Literal["session.ready"] = "session.ready"
    device_id: str
    server_time: str


class SessionError(BaseModel):
    """服务端拒绝建立会话。"""

    type: Literal["session.error"] = "session.error"
    ok: Literal[False] = False
    error: ErrorDetail

File: frontend/src/contracts/session.ts
import type { ApiError } from './envelope';

export type SessionHello = {
  type: 'session.hello';
  device_id: string;
  app_version: string;
};

export type SessionReady = {
  type: 'session.ready';
  device_id: string;
  /** Newer servers return this; the current single-user MVP server omits it. */
  user_id?: string;
  server_time: string;
};

export type SessionError = {
  type: 'session.error';
  ok: false;
  error: ApiError;
};

export type LocationReport = {
  type: 'location.report';
  schedule_scope: 'current';
  latitude: number;
  longitude: number;
  accuracy: number;
  timestamp: string;
};

export type LocationReportAck =
  | { type: 'location.report.ack'; ok: true }
  | { type: 'location.report.ack'; ok: false; error: ApiError };
